### PR TITLE
CATROID-18 user defined bricks functionality

### PR DESCRIPTION
**Pull request by BrandyMan1**
Implemented the functionality of the user defined bricks ([CATROID-18](https://jira.catrob.at/browse/CATROID-18)). The implementation makes it possible to call user scripts by using the corresponding block. Also recursive calls are possible. Inputs for user defined scripts are not implemented by this ticket (will be implemented with [CATROID-817](https://jira.catrob.at/browse/CATROID-817)).

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/EventWrapper.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ import com.badlogic.gdx.scenes.scene2d.Event
 import org.catrobat.catroid.content.eventids.EventId
 
 class EventWrapper(internal val eventId: EventId, private val wait: Boolean) : Event() {
-    private val spriteWaitList = mutableSetOf<Sprite>()
+    val spriteWaitList = mutableSetOf<Sprite>()
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/content/EventWrapper.kt:29]:
```suggestion
    private val spriteWaitList = mutableSetOf<Sprite>()
```

File: catroid/src/main/java/org/catrobat/catroid/content/EventWrapperListener.kt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
+ /*
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/content/EventWrapperListener.kt:1]:
```suggestion
/*
```

File: catroid/src/main/java/org/catrobat/catroid/content/Sprite.java [hunk 3]
```diff
@@ -346,9 +346,11 @@ public Sprite convert() {
 		return convertedSprite;
 	}
 
-	private ScriptSequenceAction createSequenceAction(Script script) {
+	public ScriptSequenceAction createSequenceAction(Script script) {
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/content/Sprite.java:349]:
```suggestion
	private ScriptSequenceAction createSequenceAction(Script script) {
```

**Review by gPathpp — CHANGES_REQUESTED** [review]
Well done! Please, fix these small issues 👍

**Review by gPathpp — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**